### PR TITLE
AFS_tab_univariate

### PR DESCRIPTION
**Pull request by aspina7**
Starting to pull together a funciton for univariate analysis. 
This will replace the previous univariate analysis function because the previous one: 
- used epitools, which didnt allow for stratified analysis. This uses epiR which allows for this. 
- wasnt tidyverse friendly. This one is. 
- had a different name, this one is moving towards our "tab" line of functions. [just a suggest on name, we can change to whatever we think is best!]

### CODE REVIEW

File: R/tab_univariate.R [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  # for "IRR" return counts and person time by exposure - NEEDS FIXING!!
+  # if (measure == "IRR") {
+  #
+  #   for (i in the_vars) {
+  #
+  #     # sum outcome and obstime by exposure
+  #     counts <- group_by(temp, .data[[i]])
+  #     counts <- summarise(counts,
+  #                         otcm = sum(.data[[i]]),
```
Comment (aspina7) [inline R/tab_univariate.R]: @zkamvar cant figure out why these arent cooperating!

File: R/tab_univariate.R [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  if (!is.data.frame(x)) {
+    stop("x must be a data frame")
+  }
+
+  if (!is.logical(pull(x, {{outcome}}))) {
+    stop("outcome must be a TRUE/FALSE variable")
+  }
+
+  # check for teach of the exposure variables that they are acceptable - NEEDS FIXING!!
```
Comment (aspina7) [inline R/tab_univariate.R]: @zkamvar not sure how to do this...

**Comment by aspina7** [comment]
@zkamvar mmm wamp wamp the IRR option is still failing i think

**Comment by aspina7** [comment]
> @zkamvar mmm wamp wamp the IRR option is still failing i think

also do we want to return as a tibble?

**Comment by zkamvar** [comment]
> @zkamvar mmm wamp wamp the IRR option is still failing i think

I'll move things into formal tests so that we can actually tell when things fail.

**Comment by aspina7** [comment]
added irr tests .... didnt want to run them - but numbers all seemed to check out when i looked

**Comment by zkamvar** [comment]
Just adding notes for myself:

There are several places where this code could be de-cluttered, most obviously in the extraction of values: For example, lines 260--272 are nearly identical to lines 368--380, with the slight difference of the indices:

image

While this is not bad per-se, it will be difficult to debug in the future. Moreover, this exact same pattern happens if there is a stratifying variable, the difference being that more rows are included and two columns are added to the ends.

**Comment by zkamvar** [comment]
I've been going about with the de-cluttering process by taking a portion of the code that looks like it could be broken up into smaller chunks and wrapping it in an internal function. 

For example, the code to calculate the incidence from the 3D table originally looked like this:

https://github.com/R4EPI/sitrep/blob/5cd332c79c142b4ce6f7ece8f4253d2dd734588c/R/tab_univariate.R#L520-L542

I've gone ahead and squished those down into this:

https://github.com/R4EPI/sitrep/blob/dce0dcbe6251e898040e10207c4b33f5a0fd4ede/R/pull_epitable.R#L114-L128

**Comment by zkamvar** [comment]
Gonna merge since it passed on appveyor

**Pull request merged**